src/mono/mono/mini/mini-amd64.c

```diff
@@ -1014,17 +1015,23 @@ get_call_info (MonoMemPool *mp, MonoMethodSignature *sig)
 		if (mono_method_signature_has_ext_callconv (sig, MONO_EXT_CALLCONV_SWIFTCALL)) {
 			MonoClass *swift_self = mono_class_try_get_swift_self_class ();
 			MonoClass *swift_error = mono_class_try_get_swift_error_class ();
+			MonoClass *swift_indirect_result = mono_class_try_get_swift_indirect_result_class ();
 			MonoClass *swift_error_ptr = mono_class_create_ptr (m_class_get_this_arg (swift_error));
 			MonoClass *klass = mono_class_from_mono_type_internal (sig->params [i]);
-			if (klass == swift_self && sig->pinvoke) {
+			if (klass == swift_indirect_result)
+				cinfo->swift_indirect_result_index = i;
+
+			if ((klass == swift_self || klass == swift_indirect_result) && sig->pinvoke) {
 				guint32 size = mini_type_stack_size_full (m_class_get_byval_arg (klass), NULL, sig->pinvoke && !sig->marshalling_disabled);
 				g_assert (size == 8);
 
 				ainfo->storage = ArgValuetypeInReg;
 				ainfo->pair_storage [0] = ArgInIReg;
 				ainfo->pair_storage [1] = ArgNone;
 				ainfo->nregs = 1;
-				ainfo->pair_regs [0] = GINT32_TO_UINT8 (AMD64_R13);
+				// The indirect result is RAX on AMD64. However, since RAX is used as a scratch register, 
+				// the R10 register is used instead and before the native call, the value is moved from R10 to RAX.
```

Comment:
```suggestion
				// The indirect result is passed in RAX on AMD64. However, since we don't prevent reg allocator to use RAX in m2n wrappers, the RAX is used to pass the function pointer.
				// We use the R10 register instead and before the native call, the value is moved from R10 to RAX.
```